File: src/main/java/com/wesker926/leetcode/algorithms/p0594/s1/Solution.java
package com.wesker926.leetcode.algorithms.p0594.s1;

import java.util.HashMap;
import java.util.Map;

/**
 * @author wesker.gh
 * @date 2021/11/20
 * @description 哈希表
 */
public class Solution {
    public int findLHS(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int n : nums) {
            map.put(n, map.getOrDefault(n, 0) + 1);
        }
        int max = 0;
        for (Map.Entry<Integer, Integer> e : map.entrySet()) {
            if (map.containsKey(e.getKey() + 1)) {
                max = Math.max(max, map.get(e.getKey() + 1) + e.getValue());
            }
        }
        return max;
    }
}

File: src/main/java/com/wesker926/leetcode/algorithms/p0594/s1/solution.go
// Package s1
// @author: wesker.gh
// @date: 2021/11/20
// @description: 哈希表
package s1

func findLHS(nums []int) int {
	m := make(map[int]int)
	for _, n := range nums {
		m[n]++
	}

	max := 0
	for n, cur := range m {
		if nxt, ok := m[n+1]; ok && cur+nxt > max {
			max = cur + nxt
		}
	}
	return max
}
